javascript/javascript1/week2/smartease.js

```diff
@@ -0,0 +1,177 @@
+// Flight booking fullname function
+console.log("**********************");
+console.log("Ex 1 : Flight booking fullname function");
+console.log("**********************");
+
+function getFullName(firstname, surname) {
+  return firstname + " " + surname;
+}
+const fullName1 = getFullName("Javascript", "class");
+const fullName2 = getFullName("React", "class");
+console.log(`fullName1:${fullName1}\nfullName2:${fullName2}`);
+
+//Formal fullname
+console.log("**********************");
+console.log("Ex 2: Formal fullname");
+console.log("**********************");
+function getFormalFullName(firstname, surname, useFormalName, gender) {
+  if (!firstname || !surname) {
+    return "Please provide your firstname and surname";
+  } else {
+    if (useFormalName === false || !useFormalName) {
+      return `${firstname} ${surname}`;
+    } else if (typeof useFormalName !== "boolean") {
+      return "please mention useFormalName as true or false";
+    } else if (useFormalName === true && gender === "male") {
+      return `Lord ${firstname} ${surname}`;
+    } else if (useFormalName === true && gender === "female") {
+      return `Lady ${firstname} ${surname}`;
+    } else {
+      return "Please provide the gender";
+    }
+  }
+}
```

Comment:
Good job! You understand the way to use `if` and `else if`. Yet, your outer `if` statement can just be the first `if` of the long `if else` chain.

There is no need to check the type of the variable `useFormalName`, but I understand your intention here.

`useFormalName === true` is the same as `useFormalName`, meaning both are booleans of `true`, you don't have to check that it's really true :)

Reply:
Thanks for the review, I have changed the If else statement order and used (useFormalName) instead (useFormalName === true). It works as the same.